tests/sources/xml/test_datacite.py

```diff
@@ -517,6 +517,246 @@ def test_get_contributors_transforms_correctly_if_fields_missing():
+    source_record = create_datacite_source_record_stub(
+        """
+        <identifier identifierType="DOI">10.7910/DVN/19PPE7</identifier>
+        <alternateIdentifiers>
+         <alternateIdentifier alternateIdentifierType="url">https://zenodo.org/record/5524465</alternateIdentifier>
+        </alternateIdentifiers>
+        <relatedIdentifiers>
+         <relatedIdentifier relatedIdentifierType="DOI" relationType="IsCitedBy">
+         10.1257/app.20150390</relatedIdentifier>
+         <relatedIdentifier relationType="IsVersionOf">10.5281/zenodo.5524464
+         </relatedIdentifier>
+         <relatedIdentifier relatedIdentifierType="ISBN"
+         relationType="IsIdenticalTo">1234567.5524464</relatedIdentifier>
+         <relatedIdentifier relatedIdentifierType="ISBN" relationType="Other">
+         1234567.5524464</relatedIdentifier>
+         <relatedIdentifier relatedIdentifierType="URL" relationType="IsPartOf">
+         https://zenodo.org/communities/astronomy-general</relatedIdentifier>
+        </relatedIdentifiers>
+        """
+    )
+    assert Datacite.get_identifiers(source_record) == [
+        Identifier(value="10.7910/DVN/19PPE7", kind="DOI"),
+        Identifier(value="https://zenodo.org/record/5524465", kind="url"),
+        Identifier(value="1234567.5524464", kind="IsIdenticalTo"),
+    ]
+
+
+def test_get_identifiers_transforms_correctly_if_fields_blank():
+    source_record = create_datacite_source_record_stub(
+        """
+        <identifier />
+        <alternateIdentifiers>
+         <alternateIdentifier />
+        </alternateIdentifiers>
+        <relatedIdentifiers>
+         <relatedIdentifier />
+        </relatedIdentifiers>
+        """
+    )
+    assert Datacite.get_identifiers(source_record) is None
+
+
+def test_get_identifiers_transforms_correctly_if_fields_missing():
+    source_record = create_datacite_source_record_stub()
+    assert Datacite.get_identifiers(source_record) is None
+
+
+def test_get_languages_success():
+    source_record = create_datacite_source_record_stub("<language>en_US</language>")
+    assert Datacite.get_languages(source_record) == ["en_US"]
+
+
+def test_get_languages_transforms_correctly_if_fields_blank():
+    source_record = create_datacite_source_record_stub("<language />")
+    assert Datacite.get_languages(source_record) is None
+
+
+def test_get_languages_transforms_correctly_if_fields_missing():
+    source_record = create_datacite_source_record_stub()
+    assert Datacite.get_languages(source_record) is None
+
+
+def test_get_links_success(datacite_record_all_fields):
+    source_record = create_datacite_source_record_stub()
+    datacite_transformer = Datacite("jpal", datacite_record_all_fields)
+    assert datacite_transformer.get_links(source_record) == [
+        Link(
+            url="https://dataverse.harvard.edu/dataset.xhtml?persistentId=abc123",
+            kind="Digital object URL",
+            text="Digital object URL",
+        )
+    ]
```

Comment:
This came up in Jonavelle's PR, the code requires instantiation for `self.source_base_url`, let me know if there's a better approach here. Also because this is essentially just wrapping `get_source_id`, I felt any edge case tests should probably happen around that method. But let me know if you disagree

Reply:
FWIW, I think it's fine as-is.  I'd like to continue discussing when class vs normal methods make sense, but feel like that might be something we could investigate wholesale later, then revisit these edge cases.

Reply:
Agreed that this should be revisited globally, I'll create an issue